test/sqllogictest/regclass.slt

```diff
@@ -0,0 +1,269 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# regclass
+
+statement OK
+create schema s
+
+statement ok
+CREATE TABLE t(i int);
+
+statement ok
+CREATE TABLE s.t(i int);
+
+statement ok
+CREATE DATABASE d;
+
+statement ok
+CREATE TABLE d.public.t();
+
+statement ok
+CREATE MATERIALIZED VIEW m AS SELECT * FROM t;
+
+statement ok
+CREATE MATERIALIZED VIEW s.m AS SELECT * FROM s.t;
+
+# OIDs are not stable, so we just validate that the output is number-like
```

Comment:
You can have stable OIDs if you start the test with a `reset-server` directive

Reply:
Thank you for this!